sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -139,6 +156,11 @@ struct _pi_context : _pi_object {
   // and destroy the pool if there are no alive events.
   ze_result_t decrementAliveEventsInPool(ze_event_pool_handle_t pool);
 
+  // Store USM allocator context(internal allocator structures)
+  // for USM shared allocations. There is 1-1 mapping between
+  // allocation context and L0 context as there is only 1 device per
+  // context.
```

Comment:
Very soon there will be added support for context with multiple devices, how would this code be affected? Can we make this code to be ready for the coming change?

Reply:
Yes, this will not work with multiple devices and require some rework.
Allocation routines require both context and device handles, so moving it to pi_device should minimize any potential changes.

Reply:
OK, please rework then. Note that the same device can be in multiple contexts too.

Reply:
The support of multiple devices was added: https://github.com/intel/llvm/pull/2343/files, though, it was tested on OpenCL only. @smaslov-intel, for the purposes of LIT tests, can Level Zero emulate multiple devices?

Reply:
Level Zero support is WIP -- it will currently raise an error if you try to create a context with multiple devices.